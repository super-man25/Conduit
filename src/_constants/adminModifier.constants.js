export const adminModifierReasonTypes = {
  weather: {
    label: 'Weather',
    value: 'Weather',
  },
  demand: {
    label: 'Calibration (Demand)',
    value: 'Demand',
  },
  teamStats: {
    label: 'Team Performance',
    value: 'TeamStats',
  },
  opponentStats: {
    label: 'Strength Of Opponent',
    value: 'OpponentStats',
  },
  other: {
    label: 'Other',
    value: 'Other',
  },
};
