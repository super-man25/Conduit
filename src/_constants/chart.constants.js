import { cssConstants } from '.';

export const chartLabelStyles = {
  fontSize: 12,
  color: cssConstants.PRIMARY_GRAY
};

export const GROUP_FILTERS = {
  periodic: 0,
  cumulative: 1
};

export const CHART_HEIGHT = 400;
export const DAILY = 1;
export const END_OF_DAY = 24;
export const WEEKLY = 7;
export const BI_WEEKLY = 14;
export const MONTHLY = 1;
export const END_OF_MONTH = 31;
export const QUARTERLY = 3;
export const BUSINESS_WEEK = 5;
