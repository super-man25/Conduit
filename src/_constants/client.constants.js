export const clientConstants = {
  GET_REQUEST: 'CLIENT_GET_REQUEST',
  GET_SUCCESS: 'CLIENT_GET_SUCCESS',
  GET_FAILURE: 'CLIENT_GET_FAILURE',

  UPDATE_REQUEST: 'CLIENT_UPDATE_REQUEST',
  UPDATE_SUCCESS: 'CLIENT_UPDATE_SUCCESS',
  UPDATE_FAILURE: 'CLIENT_UPDATE_FAILURE'
};
