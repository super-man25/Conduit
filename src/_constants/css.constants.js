export const cssConstants = {
  PRIMARY_EVEN_LIGHTER_GRAY: '#F2F2F2',
  PRIMARY_DARK_BLUE: '#024A8F',
  PRIMARY_LIGHT_BLUE: '#38A9DB',
  PRIMARY_LIGHTEST_GRAY: '#F8F8F8',
  PRIMARY_WHITE: '#FFFFFF',
  PRIMARY_LIGHTER_GRAY: '#EEEEEE',
  PRIMARY_LIGHT_GRAY: '#D1D1D1',
  PRIMARY_GRAY: '#999999',
  PRIMARY_DARK_GRAY: '#767676',
  PRIMARY_DARKEST_GRAY: '#323232',
  PRIMARY_LIGHT_BLACK: '#070707',
  SECONDARY_BLUE_BLACK: '#031E35',
  SECONDARY_BLUE: '#2670AE',
  SECONDARY_LIGHT_BLUE: '#4B98CF',
  SECONDARY_LIGHTEST_BLUE: '#7CBCDD',
  SECONDARY_BLUE_GRAY: '#C4D2E1',
  SECONDARY_PURPLE: '#621F95',
  SECONDARY_LIGHT_PURPLE: '#817CB8',
  SECONDARY_LIGHTER_PURPLE: '#BDA3D1',
  SECONDARY_BURNT_ORANGE: '#F3A712',
  SECONDARY_YELLOW: '#FDE74C',
  SECONDARY_RED: '#DC0A0A',
  SECONDARY_GREEN: '#399E5A',
  SECONDARY_LIGHT_GREEN: '#9BC53D',
  TITLE_SIZE_H1: '36px',
  TITLE_WEIGHT_H1: 'normal',
  TITLE_SIZE_H2: '28px',
  TITLE_WEIGHT_H2: 'normal',
  TITLE_SIZE_H3: '22px',
  TITLE_WEIGHT_H3: 'normal',
  TITLE_SIZE_H4: '18px',
  TITLE_WEIGHT_H4: 'normal',
  TITLE_SIZE_H5: '14px',
  TITLE_WEIGHT_H5: 'bold',
  PARAGRAPH_SIZE_P1: '36px',
  PARAGRAPH_WEIGHT_P1: 'bold',
  PARAGRAPH_SIZE_P2: '28px',
  PARAGRAPH_WEIGHT_P2: 'normal',
  SUBHEADING_SIZE_S1: '12px',
  SUBHEADING_WEIGHT_S1: 'normal',
  LABEL_SIZE: '14px',
  LABEL_WEIGHT: 'normal'
};

export const shadows = {
  SHADOW_PRIMARY: '0 1px 1px 0 #BBBBBB, 0 2px 4px 0 rgba(0,0,0,0.5);'
};

export const zIndexes = {
  BASE: '1000'
}
