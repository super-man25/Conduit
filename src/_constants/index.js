export * from './alert.constants';
export * from './user.constants';
export * from './css.constants';
export * from './url.constants';
