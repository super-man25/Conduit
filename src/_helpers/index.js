export * from './browser-utils';
export * from './chart-utils';
export * from './date-utils';
export * from './history';
export * from './js-utils';
export * from './pricing-utils';
export * from './string-utils';
export * from './style-utils';
export * from './svg-utils';
