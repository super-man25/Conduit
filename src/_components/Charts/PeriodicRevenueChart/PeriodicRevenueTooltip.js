// @flow
import * as React from 'react';
import {
  ChartTooltip,
  TooltipHeaderText,
  TooltipBodyTitle,
  TooltipBodyText
} from '../ChartTooltip';
import { formatUSD } from '_helpers/string-utils';

const CHART_KEYS = {
  actual: 'periodicRevenue',
  projected: 'projectedPeriodicRevenue'
};

type TooltipContentProps = {
  active?: boolean,
  payload?: any,
  dateFormatter: (Date) => string
};

export function PeriodicRevenueTooltip(props: TooltipContentProps) {
  const { payload, active, dateFormatter } = props;

  if (!active || !payload || !payload.length) {
    return null;
  }

  const dataPoint = payload.length && payload[0].payload;
  const bodyTitle = dataPoint.isProjected ? 'Projected Revenue' : 'Revenue';
  const bodyText = formatUSD(
    dataPoint.isProjected
      ? dataPoint[CHART_KEYS.projected]
      : dataPoint[CHART_KEYS.actual]
  );

  const header = (
    <TooltipHeaderText>{dateFormatter(dataPoint.timestamp)}</TooltipHeaderText>
  );

  const body = (
    <React.Fragment>
      <TooltipBodyTitle>{bodyTitle}</TooltipBodyTitle>
      <TooltipBodyText>{bodyText}</TooltipBodyText>
    </React.Fragment>
  );

  return <ChartTooltip headerComponent={header} bodyComponent={body} />;
}
