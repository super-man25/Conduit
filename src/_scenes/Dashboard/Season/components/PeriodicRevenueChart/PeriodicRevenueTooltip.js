// @flow
import * as React from 'react';
import {
  ChartTooltip,
  TooltipHeaderText,
  TooltipBodyTitle,
  TooltipBodyText
} from '../ChartTooltip';
import { format } from 'date-fns';
import { formatUSD } from '_helpers/string-utils';
import { DATE_FORMATS } from '_constants';

const CHART_KEYS = {
  actual: 'periodicRevenue',
  projected: 'projectedPeriodicRevenue'
};

type TooltipContentProps = {
  active?: boolean,
  payload?: any,
  dateFormat: string
};

export function PeriodicRevenueTooltip(props: TooltipContentProps) {
  const { payload, active, dateFormat } = props;

  if (!active || !payload || !payload.length) {
    return null;
  }

  const dataPoint = payload.length && payload[0].payload;
  const bodyTitle = dataPoint.isProjected ? 'Projected Revenue' : 'Revenue';
  const bodyText = formatUSD(
    dataPoint.isProjected
      ? dataPoint[CHART_KEYS.projected]
      : dataPoint[CHART_KEYS.actual]
  );

  const header =
    dateFormat === DATE_FORMATS.day ? (
      <TooltipHeaderText>
        {format(dataPoint.timestamp, dateFormat)}
      </TooltipHeaderText>
    ) : (
      <React.Fragment>
        <TooltipHeaderText>
          {format(dataPoint.timestamp, DATE_FORMATS.day)}
        </TooltipHeaderText>
        <TooltipHeaderText>
          {format(dataPoint.timestamp, DATE_FORMATS.time)}
        </TooltipHeaderText>
      </React.Fragment>
    );

  const body = (
    <React.Fragment>
      <TooltipBodyTitle>{bodyTitle}</TooltipBodyTitle>
      <TooltipBodyText>{bodyText}</TooltipBodyText>
    </React.Fragment>
  );

  return <ChartTooltip headerComponent={header} bodyComponent={body} />;
}
