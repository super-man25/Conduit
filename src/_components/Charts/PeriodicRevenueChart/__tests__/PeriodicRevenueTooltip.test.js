import { shallow } from 'enzyme';
import React from 'react';
import { READABLE_DATETIME_FORMAT } from '_constants';
import { dateFormatter } from '_helpers';
import { PeriodicRevenueTooltip } from '../PeriodicRevenueTooltip';

const createProps = () => ({
  active: true,
  payload: [
    {
      payload: {
        eventId: 1,
        id: 1,
        inventory: 100,
        periodicInventory: 100,
        revenue: 100,
        periodicRevenue: 100,
        isProjected: false,
        timestamp: 1529336865706
      }
    }
  ],
  dateFormatter: (d) => d.toString()
});

describe('<PeriodicRevenueTooltip />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<PeriodicRevenueTooltip {...createProps()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when active is false', () => {
    const props = { ...createProps(), active: false };
    const wrapper = shallow(<PeriodicRevenueTooltip {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when no payload is passed', () => {
    const props = { ...createProps(), payload: [] };
    const wrapper = shallow(<PeriodicRevenueTooltip {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly with different dateformats ', () => {
    const props = {
      ...createProps(),
      dateFormatter: dateFormatter(READABLE_DATETIME_FORMAT, 'America/New_York')
    };
    const wrapper = shallow(<PeriodicRevenueTooltip {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
