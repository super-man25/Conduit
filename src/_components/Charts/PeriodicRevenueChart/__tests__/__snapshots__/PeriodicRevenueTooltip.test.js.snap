// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PeriodicRevenueTooltip /> should render correctly 1`] = `
<ChartTooltip
  bodyComponent={
    <React.Fragment>
      <ForwardRef>
        Revenue
      </ForwardRef>
      <ForwardRef>
        $100.00
      </ForwardRef>
    </React.Fragment>
  }
  headerComponent={
    <ForwardRef>
      2018-06-18T15:47:45.706Z @MMM DD, YYYY
    </ForwardRef>
  }
/>
`;

exports[`<PeriodicRevenueTooltip /> should render correctly when active is false 1`] = `""`;

exports[`<PeriodicRevenueTooltip /> should render correctly when no payload is passed 1`] = `""`;

exports[`<PeriodicRevenueTooltip /> should render correctly with different dateformats  1`] = `
<ChartTooltip
  bodyComponent={
    <React.Fragment>
      <ForwardRef>
        Revenue
      </ForwardRef>
      <ForwardRef>
        $100.00
      </ForwardRef>
    </React.Fragment>
  }
  headerComponent={
    <React.Fragment>
      <ForwardRef>
        2018-06-18T15:47:45.706Z @MMM DD, YYYY
      </ForwardRef>
      <ForwardRef>
        2018-06-18T15:47:45.706Z @hh:mm a
      </ForwardRef>
    </React.Fragment>
  }
/>
`;
